### PR TITLE
Refactor OneDimensionalCover to avoid boolean mask repetitions

### PR DESCRIPTION
**Pull request by ulupo**
#### What does this implement/fix? Explain your changes.
- Add user convenience method to OneDimensionalCover to produce interval limits
- Make interval limits attributes learned in fit and used in transform
- Avoid perfect repetitions of boolean masks (i.e. identical pullback cover sets) while respecting the ascending order of filtration values
- Rewrote test for shapes as now one cannot expect as many columns (masks) as intervals at initialisation.

Any of these suggested changes, if accepted, must then also be implemented as appropriate in CubicalCover!

### CODE REVIEW

**Comment by lewtun** [comment]
Will accepting these changes _break_ `CubicalCover` or is it simply something we need to add for consistency?

File: giotto/mapper/cover.py [hunk 1]
```diff
@@ -10,16 +11,29 @@ def __init__(self, n_intervals=10, overlap_frac=0.1):
...
         if X.ndim == 1:
             X = X[:, None]
 
         if X.size == 1:
             return np.array([[True]])
 
+        Xt = (X > self.left_limits_) & (X < self.right_limits_)
+        # Avoid repeating the same boolean masks (columns)
+        Xt, indices = np.unique(Xt, axis=1, return_index=True)
+        # Respect the order in left_limits and right_limits_
```
Comment (lewtun) [inline giotto/mapper/cover.py]: Add trailing underscore to `left_limits`

**Comment by ulupo** [comment]
As far as I can tell, CubicalCover should remain intact and its interaction with downstream tasks unchanged.

File: giotto/mapper/cover.py [hunk 2]
```diff
@@ -29,12 +43,14 @@ def transform(self, X, y=None):
...
-        right_endpts[-1] = np.inf
-        Xt = (X > left_endpts) & (X < right_endpts)
-        return Xt
+        left_limits = np.linspace(m, last, num=self.n_intervals, endpoint=True)
+        right_limits = left_limits + l
+        left_limits[0] = -np.inf
+        right_limits[-1] = np.inf
+        return left_limits, right_limits
+
+    def find_intervals(self, X):
```
Comment (lewtun) [inline giotto/mapper/cover.py]: Is this function used anywhere? Remove if not?

File: giotto/mapper/cover.py (hunk 1 above)
Comment (ulupo) [reply giotto/mapper/cover.py]: Just to understand: you mean in the comments?

File: giotto/mapper/cover.py (hunk 2 above)
Comment (ulupo) [reply giotto/mapper/cover.py]: This is the first item in the description of this pull request.  I imagined some more technical users might find this method useful for debugging. Agree/disagree? We could always make it hidden.

Comment (lewtun) [reply giotto/mapper/cover.py]: Just add a small docstring so we remember what we wanted to use this for in future

File: giotto/mapper/cover.py (hunk 1 above)
Comment (ulupo) [reply giotto/mapper/cover.py]: I assumed you did and pushed the fix.

File: giotto/mapper/cover.py (hunk 2 above)
Comment (ulupo) [reply giotto/mapper/cover.py]: Renamed and better documented.

**Pull request merged**